test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -290,6 +313,220 @@ test.describe( 'Navigation block', () => {
 		await expect( warningMessage ).toBeVisible();
 	} );
 
+	test( 'creating navigation menus via keyboard without losing focus', async ( {
```

Comment:
This test is huge, but it's better that it is:
- Faster than running it in several smaller tests
- We don't need to "redo" work in each test. For example, each test needs to build off the previous one, so there would either be repeating tests or lots of initial menus to start from.
- Creating a navigation block using a keyboard should work. This is a more realistic example of creating a menu and is more likely to catch edge cases.

I tried to break it down into sensible smaller tests, and refactor reusable pieces but keep them readable. I wrote this with the idea that it will likely need to be debugged since I expect focus management bugs to happen. When we debug it or change the implementation, my intent is that this will be easier to understand and maintain.